### ISSUE TITLE
Is it possible to customise the base size and layout of widgets in the grid?

### ISSUE THREAD

**Issue by jameswtelfer** [gottfrois/dashing-rails#102]
With the original dashing it is possible to customise the size & layout of the widget grid via the [application.coffee](https://github.com/Shopify/dashing/blob/master/templates/project/assets/javascripts/application.coffee) script that is generated for each dashboard.

I tried lifting the version out of the dashing-rails gem and replicating within my application assets with the new values, which worked so long as `config.assets.debug = true` is present within the environment config, which obviously means it doesn't work in production.  In production the updated code is present within `application.js`, but does not get executed, instead the code from the gem is executed.

Any suggestions appreciated as I've reached the limit of my abilities at this point!

**Comment by gottfrois** [comment]
have you tried to override this file? https://github.com/gottfrois/dashing-rails/blob/master/app/assets/javascripts/dashing/dashing.coffee

On your rails app, create an exact copy of this file, located at the same location and add your custom code.

Please let me know.

**Comment by jameswtelfer** [comment]
Oddly, that works in the development environment, but not in production.  Looking at the generated application.js it looks like the overridden version of the file is not included in the compiled assets.

**Comment by gottfrois** [comment]
Have you tried adding it manually in your `application.rb`?

**Comment by jameswtelfer** [comment]
That does something different again.  The new code _is_ included in the generated js, but never gets executed (I'm basing that on changing the text that gets printed to the JS console, plus the widgets don't change size.  Works as expected in development environment.

**Comment by gottfrois** [comment]
That's progress :)
It's weird though that if the JS file is included it does not get executed :-/

**Comment by jameswtelfer** [comment]
Agree it's weird, will have a poke at it over the weekend.  It's a toy project at present so if I end up running it in development env then so be it!

**Comment by gottfrois** [comment]
yeah, to be honest, dashing rails is not really production ready due to using the rails action live...

**Comment by NashL** [comment]
Hi @jameswtelfer, did you solve this problem? I'm trying to customize the size at production too

**Comment by jameswtelfer** [comment]
@NashL not really I'm afraid.  It sort of worked if I tuned on all the options to enable the asset generation pipeline in production, but at that point you're practically in development anyway.

**Comment by gottfrois** [comment]
the ideal approach would be to refactor the project using action cable instead
